ring_buffer/ring_buffer.py

```diff
@@ -8,25 +8,82 @@ def __init__(self, capacity):
         self.storage = DoublyLinkedList()
 
     def append(self, item):
-        pass
+        if self.storage.length == 0:
+            self.storage.add_to_head(item)
+            self.current = self.storage.head
+            #nothing in DLL, add item to start
+        elif self.storage.length < self.capacity:
+            self.storage.add_to_tail(item)
+            self.current = self.storage.tail
+            #still have room in DLL, add item to list
+        elif self.current is self.storage.tail:
+            self.storage.remove_from_head()
+            self.storage.add_to_head(item)
+            self.current = self.storage.head
+            # if current is the last item in list
+            # remove item from start
+            # add this new item to the start
+            # now current is the head of the DLL
+        else:
+            self.current.insert_after(item)
```

Comment:
This is the only area I think you might have a small bug; you might be exceeding the capacity with the self.storage.length method. Need to experiment, but since it passes the tests, this is fine! Great work!